src/backend/joanie/core/models/courses.py

```diff
@@ -365,3 +386,33 @@ def clean(self):
                     )
 
         super().clean()
+
+    def get_synchronize_course_run_data(self, hidden=False):
+        """
+        Return data for the course run that will be sent to the remote web hooks.
+        If the course run is hidden, we set the dates to None for the synchronization.
+        """
+        site = Site.objects.get_current()
+        resource_path = reverse("course-runs-detail", kwargs={"id": self.id})
+        return {
+            "catalog_visibility": enums.COURSE_AND_SEARCH
+            if self.is_listed and not hidden
+            else enums.HIDDEN,
+            "course": self.course.code,
+            "end": self.end if not hidden else None,
+            "enrollment_start": self.enrollment_start if not hidden else None,
+            "enrollment_end": self.enrollment_end if not hidden else None,
+            "languages": self.languages,
+            "resource_link": f"https://{site.domain:s}{resource_path:s}",
+            "start": self.start if not hidden else None,
+        }
+
+    @staticmethod
+    def synchronize_course_run(course_run, hidden=False):
+        """
+        Synchronize a course run by calling remote web hooks.
+        """
+
+        SynchronizeHook.submit_hook(
```

Comment:
this should be in `synchronize_with_webhooks` as we want all synchronization to be done in this method